machine/machine_core/ssh_connection.py

```diff
@@ -417,16 +417,23 @@ def write(self, dest, content, append=False, owner=None, perm=None):
             append: If True, append to existing file instead of replacing it
             owner: If set, call chown on the file with the given owner string
             perm: Optional file permission as chmod shell string (e.g. "0600")
+
+        The directory of dest is created automatically.
         """
         assert dest
         assert self.ssh_address
 
-        cmd = "cat %s '%s'" % (append and '>>' or '>', dest)
+        qdest = shlex.quote(dest)
```

Comment:
wwg1wga

Reply:
Not sure what that means, but if it's a complaint about my abbrev, I renamed it to `dest_quoted`.